### PR TITLE
Make onboarding aws account and Budget feature workable in china regions

### PR DESCRIPTION
**Pull request by dengmingtong**
…ngxia regions

Issue #, if available:

Description of changes:

Checklist:

- [ yes] Have you successfully deployed to an AWS account with your changes?
- [ yes] Have you successfully deployed to an AWS China account with your changes?
- [ yes] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully tested with your changes locally?
- [ ] If new dependencies have been added, have they been pinned to specific versions?
- [ ] Is this change also required on the AWS Solution version?
- [ ] Have you updated openapi.yaml if you made updates to API definition (including add, delete or update parameter and request data schema)?
- [ ] If you had to run manual tests, have you considered automating those tests by adding them to [end-to-end tests](../main/end-to-end-tests/README.md)?
- [ ] If you are updating the changelog and vending out a new release, have you updated versionNumber and versionDate in [.defaults.yml](../main/config/settings/.defaults.yml)

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml [hunk 1]
```diff
@@ -751,14 +751,7 @@ Resources:
...
             Action:
-              - 'kms:DescribeKey'
-              - 'kms:Encrypt'
-              - 'kms:Decrypt'
-              - 'kms:ReEncrypt*'
-              - 'kms:GenerateDataKey'
-              - 'kms:GenerateDataKeyWithoutPlaintext'
-              - 'kms:CreateGrant'
-              - 'kms:RevokeGrant'
+              - 'kms:*'
```
Comment (zxkane) [inline addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml]: why changing to all actions?

File: addons/addon-base-raas/packages/base-raas-services/lib/aws-accounts/__tests__/aws-cfn-service.test.js [hunk 2]
```diff
@@ -26,9 +26,6 @@ const AuthServiceMock = require('@amzn/base-services/lib/authorization/authoriza
 jest.mock('@amzn/base-services/lib/audit/audit-writer-service');
 const AuditServiceMock = require('@amzn/base-services/lib/audit/audit-writer-service');
 
-jest.mock('@amzn/base-services/lib/settings/env-settings-service');
-const SettingsServiceMock = require('@amzn/base-services/lib/settings/env-settings-service');
```
Comment (zxkane) [inline addons/addon-base-raas/packages/base-raas-services/lib/aws-accounts/__tests__/aws-cfn-service.test.js]: looks like you are using different mocking data for different test cases?

is it possible to register the mocking `SettingService` in the targeted test cases?

File: addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js [hunk 3]
```diff
@@ -225,13 +233,26 @@ class BudgetsService extends Service {
...
-      secretAccessKey,
-      sessionToken,
-    });
+    const region = this.settings.get(settingKeys.awsRegion);
+    let budgetsClient;
+    if (region === 'cn-northwest-1' || region === 'cn-north-1') {
+      budgetsClient = new aws.sdk.Budgets({
+        apiVersion: '2017-10-25',
+        region,
+        endpoint: 'budgets.amazonaws.com.cn',
```
Comment (zxkane) [inline addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js]: if it's required, pls create variables for endpoint and region, moving budgetClient out of if...else clause

File: addons/addon-base-raas/packages/base-raas-services/lib/helpers/utils.js [hunk 4]
```diff
@@ -189,6 +189,13 @@ function removeAccountFromStatement(oldStatement, memberAccountId) {
   return statement;
 }
 
+function getAwsConsoleSuffixByRegion(region) {
+  if (region === 'cn-north-1' || region === 'cn-northwest-1') {
```
Comment (zxkane) [inline addons/addon-base-raas/packages/base-raas-services/lib/helpers/utils.js]: prefer to `region.startsWith('cn-')`

File: addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js [hunk 5]
```diff
@@ -173,13 +176,18 @@ class BudgetsService extends Service {
   }
 
   _formBudgetObject(newBudgetConfig) {
+    const region = this.settings.get(settingKeys.awsRegion);
+    let unit = 'USD';
+    if (region === 'cn-north-1' || region === 'cn-northwest-1') {
```
Comment (zxkane) [inline addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js]: region.startsWith

File: addons/addon-base/packages/services-container/lib/service.js [hunk 6]
```diff
@@ -59,6 +59,7 @@ const Boom = require('./boom');
 const settingKeys = {
   awsSuffix: 'awsSuffix',
   awsPartition: 'awsPartition',
+  awsRegion: 'awsRegion',
```
Comment (zxkane) [inline addons/addon-base/packages/services-container/lib/service.js]: why not using partition?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml (hunk 1 above)
Comment (dengmingtong) [reply addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml]: The cloudformation was failed before changed to 'kms:*', below is error message:
`The new key policy will not allow you to update the key policy in the future`

But I deep dived again, I found this is caused by my [commit](https://github.com/awslabs/service-workbench-on-aws-cn/commit/01e6ce53a4b1e12c7108b0aa7b3a2834acc9faf7) , which change AWS::AccountId to CentralAccountId, this caused the accounts are different when we add the external aws account.  
I changed it back. 
Thank you very much to help find this issue.

File: addons/addon-base-raas/packages/base-raas-services/lib/aws-accounts/__tests__/aws-cfn-service.test.js (hunk 2 above)
Comment (dengmingtong) [reply addons/addon-base-raas/packages/base-raas-services/lib/aws-accounts/__tests__/aws-cfn-service.test.js]: Yes, there are two ways to implementation to mock settings. 
Recoverd to SettingsServiceMock way.

File: addons/addon-base/packages/services-container/lib/service.js (hunk 6 above)
Comment (dengmingtong) [reply addons/addon-base/packages/services-container/lib/service.js]: it's useless, forgot deleting.

File: addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js (hunk 5 above)
Comment (dengmingtong) [reply addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js]: Done

File: addons/addon-base-raas/packages/base-raas-services/lib/helpers/utils.js (hunk 4 above)
Comment (dengmingtong) [reply addons/addon-base-raas/packages/base-raas-services/lib/helpers/utils.js]: Done

File: addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js (hunk 3 above)
Comment (dengmingtong) [reply addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js]: Yes, refactored

**Review by zxkane — APPROVED** [review]

**Pull request merged**